File: base/src/main/vsdk/toolkit/common/_AlgebraicExpressionVariableNode.java
package vsdk.toolkit.common;
import java.io.Serial;

public class _AlgebraicExpressionVariableNode extends _AlgebraicExpressionNode
{
    @Serial private static final long serialVersionUID = 20071014L;

    private AlgebraicExpression parent;
    private String name;

    public _AlgebraicExpressionVariableNode(AlgebraicExpression parent, String name)
    {
        this.parent = parent;
        this.name = name;
    }

    @Override
    public double eval() throws AlgebraicExpressionException
    {
        double val = parent.getVariableValue(name);
        return val;
    }

    @Override
    public String toString()
    {
        String msg;

        msg = name;
        return msg;
    }

}

File: typescript/base/src/main/vsdk/toolkit/common/symbolicAlgebra/_AlgebraicExpressionVariableNode.ts
import { _AlgebraicExpressionNode } from "./_AlgebraicExpressionNode.js";
import type { AlgebraicExpression } from "./AlgebraicExpression.js";
export class _AlgebraicExpressionVariableNode extends _AlgebraicExpressionNode {
    public constructor(
        private readonly parent: AlgebraicExpression,
        private readonly name: string,
    ) {
        super();
    }
    public eval(): number {
        return this.parent.getVariableValue(this.name);
    }
    public override toString(): string {
        return this.name;
    }
}
